rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,272 @@
+# RFC Template
+
+- Feature Name: `single-cluster-dnspolicy`
+- Start Date: 2023-10-09
+- RFC PR: [Kuadrant/architecture#30](https://github.com/Kuadrant/architecture/pull/30)
+- Issue tracking: [Kuadrant/architecture#31](https://github.com/Kuadrant/architecture/issues/31)
+
+# Summary
+[summary]: #summary
+
+Proposal for changes to the `DNSPolicy` API to allow it to provide a simple dns strategy as an option in a single cluster context. This will remove, but not negate, the complex DNS structure we use in a multi-cluster environment and in doing so allow use of popular dns integrators such as external-dns .
+
+# Motivation
+[motivation]: #motivation
+
+The [`DNSPolicy`](https://github.com/Kuadrant/multicluster-gateway-controller/blob/main/pkg/apis/v1alpha1/dnspolicy_types.go) API (v1alpha1), was implemented as part of our multi cluster gateway offering using OCM and as such the design and implementation were influenced heavily by how we want multi cluster dns to work.
+
+* Decouple the API entirely from OCM and multi cluster specific concepts.
+* Simplify the DNS record structure created for a gateway listeners host for single cluster use.
+* Improve the likelihood of adoption by creating an integration path for other kubernetes dns controllers such as [external-dns](https://github.com/kubernetes-sigs/external-dns).
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The DNSPolicy can be used to target a Gateway in a single cluster context and will create dns records for each listener host in an appropriately configured external dns provider.
+In this context the advanced `loadbalancing` configuration is unnecessary, and the resulting DNSRecord can be created mapping individual listener hosts to a single DNS A or CNAME record by using the `simple` strategy in the DNSPolicy.
+
+**Example 1.** DNSPolicy using `simple` strategy
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: my-route53-credentials
+    namespace: my-gateways
+    kind: Secret
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  strategy: simple
```

Comment:
been thinking about this property. WDYT to changing it to be `routingStrategy` might convey a little better what it does?